### ISSUE TITLE
Load alternate configuration on unsupported versions

### ISSUE THREAD

**Issue by rjorgenson** [gpakosz/.tmux#473]
I am attempting to setup a tmux config that will work on various tmux versions. My first attempt was to load configuration conditionally in my `~/.tmux.conf` file based on the tmux version, which works great but .tmux doesn't like not being at `~/.tmux.conf` .. is there any way around this? Maybe adding a `~/.tmux.conf.unsupported` (similar to `~/.tmux.conf.local`) option that gets loaded instead of .tmux when it detects unsupported versions?

**Comment by gpakosz** [comment]
Hello @rjorgenson 👋

I'm not sure I understand what you're trying to achieve. If you're using tmux < 2.3 then you're really better of not using Oh my tmux! at all. After all it's just a matter of using another `.tmux.conf` file.

**Comment by rjorgenson** [comment]
@gpakosz That is essentially the goal, but I deploy my dotfiles to quite a few systems and right now I have to do that logic at deploy time and put the right `.tmux.conf` in place, and then if that system ever changes down the line I have to manually put .tmux's `.tmux.conf` into place to start using it. The goal is to have a single configuration file that works on all systems and loads .tmux if it is supported. My initial attempt was to use a `.tmux.conf` that detected what tmux version I had and loaded configurations accordingly but .tmux doesn't work if it's not your `~/.tmux.conf` file.

**Comment by gpakosz** [comment]
I don't know how to address that. I would find a way to deploy tmux > 2.3 (which was released in 2016) on the different hosts 🙂

**Comment by rjorgenson** [comment]
These aren't typically hosts I have control of. Would it not be possible to detect the version just like you're doing now, and instead of killing the server load the alternate config and only load the rest of the config if the version is supported?

**Comment by gpakosz** [comment]
I don't think this is a thing I want to support.

But in the meantime you can try this `~/.tmux.conf` (still assuming Oh my tmux! has been cloned in `~/.tmux`)
```
# : << EOF

# put your base tmux config here
set -g default-terminal "screen-256color" # colors!
setw -g xterm-keys on
set -s escape-time 10                     # faster command sequences
set -sg repeat-time 600                   # increase repeat timeout
set -s focus-events on

# -- 8< ------------------------------------------------------------------------

run 'cut -c3- ~/.tmux.conf | sh -s _oh_my_tmux_if_supported'

# EOF
#
# # exit the script if any statement returns a non-true return value
# set -e
#
# unset GREP_OPTIONS
# export LC_NUMERIC=C
#
# _tmux_version=$(tmux -V | awk '{gsub(/[^0-9.]/, "", $2); print ($2+0) * 100}')
#
# _oh_my_tmux_if_supported() {
#   if [ $_tmux_version -ge 230 ]; then
#     tmux source ~/.tmux/.tmux.conf
#   fi
# }
#
# if [ x"$@" = x"_oh_my_tmux_if_supported" ]; then
#   _oh_my_tmux_if_supported
# else
#   cut -c3- ~/.tmux/.tmux.conf | sh -s "$@"
# fi
```

**Comment by rjorgenson** [comment]
I'll give this a shot when I get a chance, but at first look this doesn't look any different than what I was trying to do, when `~/.tmux/.tmux.conf` is sourced it will attempt to `cut` all of it's bit's out of `~/.tmux.conf` but that's not where that file lives. This is the initial problem I ran into when trying to do a conditional load of .tmux.

**Comment by gpakosz** [comment]
To make things clear you should clone Oh my tmux! into `~/.tmux` and instead of a `~/.tmux.conf → ~/.tmux/.tmux.conf` symlink you should create a file named `~/.tmux.conf`with the content given above 👆

That new `~/.tmux.conf` file will act as a trampoline to `~/.tmux/.tmux.conf` is tmux version is >= 2.3

**Comment by rjorgenson** [comment]
Yes I get that, but when I tried to do that previously (granted in a more basic way, and I don't entirely understand everything that's going on in there but it doesn't appear like it would do anything differently than I was doing) this line https://github.com/gpakosz/.tmux/blob/760bf4fabda9a41f67f8ad7e202c2a66a2a3d4e1/.tmux.conf#L170 would not run because none of the code it's trying to pull out of `~/.tmux.conf` and run exists in that file because all it has is the conditional sourcing.

**Comment by rjorgenson** [comment]
After a few attempts at melting my brain trying to understand what that was doing it finally clicked! This works perfectly for my purposes, thanks!

**Comment by rjorgenson** [comment]
@gpakosz Running into an issue with this solution, can you think of any reason why TPM would not be loading plugins with this config file but does with the original oh-my-tmux `.tmux.conf` in place? Sourcing from the same `.tmux.conf.local` with the below tpm settings in it? Running `~/.tmux/plugins/tmux-resurrect/resurrect.tmux` after the session is up loads the plugin and I am able to restore and save my sessions but it's not loaded at runtime and continuum is also not loaded so sessions aren't saved. Everything works as expected when using the original `.tmux.conf` instead of this wrapper solution.

```
# -- tpm -----------------------------------------------------------------------

# while I don't use tpm myself, many people requested official support so here
# is a seamless integration that automatically installs plugins in parallel

# whenever a plugin introduces a variable to be used in 'status-left' or
# 'status-right', you can use it in 'tmux_conf_theme_status_left' and
# 'tmux_conf_theme_status_right' variables.

# by default, launching tmux will update tpm and all plugins
#   - true (default)
#   - false
tmux_conf_update_plugins_on_launch=true

# by default, reloading the configuration will update tpm and all plugins
#   - true (default)
#   - false
tmux_conf_update_plugins_on_reload=true

# /!\ do not add set -g @plugin 'tmux-plugins/tpm'
# /!\ do not add run '~/.tmux/plugins/tpm/tpm'

# to enable a plugin, use the 'set -g @plugin' syntax:
# visit https://github.com/tmux-plugins for available plugins
#set -g @plugin 'tmux-plugins/tmux-copycat'
#set -g @plugin 'tmux-plugins/tmux-cpu'
#set -g @plugin 'tmux-plugins/tmux-resurrect'
#set -g @plugin 'tmux-plugins/tmux-continuum'
#set -g @continuum-restore 'on'

# TPM configuration
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @plugin 'tmux-plugins/tmux-continuum'

set -g @continuum-restore 'on'
set -g @continuum-save-interval '15'
```

**Comment by gpakosz** [comment]
Plugins are not loading because TPM inspects `~/.tmux.conf` and follows `source` commands.

You can try:
```
# : << EOF

# put your base tmux config here
set -g default-terminal "screen-256color" # colors!
setw -g xterm-keys on
set -s escape-time 10                     # faster command sequences
set -sg repeat-time 600                   # increase repeat timeout
set -s focus-events on

source -q ~/.tmux.conf.local

# -- 8< ------------------------------------------------------------------------

run 'cut -c3- ~/.tmux.conf | sh -s _oh_my_tmux_if_supported'

# EOF
#
# # exit the script if any statement returns a non-true return value
# set -e
#
# _tmux_version=$(tmux -V | awk '{gsub(/[^0-9.]/, "", $2); print ($2+0) * 100}')
#
# _oh_my_tmux_if_supported() {
#   if [ $_tmux_version -ge 230 ]; then
#     tmux source ~/.tmux/.tmux.conf
#   fi
# }
#
# if [ x"$@" = x"_oh_my_tmux_if_supported" ]; then
#   _oh_my_tmux_if_supported
# else
#   cut -c3- ~/.tmux/.tmux.conf | sh -s "$@"
# fi
```

**Comment by gpakosz** [comment]
@rjorgenson Hello 👋

Did you give it a try?

**Comment by rjorgenson** [comment]
@gpakosz sorry I thought I responded to this already. That did work for enabling the plugins in versions >=2.3 but it sourced the local config twice and sourced the local config on all versions. I ended up just reverting to using a different .tmux.conf on hosts with tmux <2.3 as it seems there's just not a good way to accomplish that. Thanks for all your help though! I even learned a bit about the voodoo that you do to get this to work =]

**Comment by gpakosz** [comment]
Hmm then I think what you could do is put the `set -g plugin ...` in the main `.tmux.conf` instead of `source -q ~/.tmux.conf.local`. This is the answer I should have made in fact!

**Comment by rjorgenson** [comment]
@gpakosz I tested that out and it did work but has the side effect of doing a continuum restore every time I reload the config when not using dottmux, I was not able to figure out why it does this. I tried putting various pieces in different stages in the config but no matter where I put it when the config was reloaded and the `_oh_my_tmux_if_supported` function call loaded the unsupported tmux config file it would run a continuum restore. Manually restoring is definitely not a deal breaker for me as most machines hardly ever go down. Think i'll go with this approach, thanks!
